### PR TITLE
Card Metadata Implementation

### PR DESCRIPTION
**Pull request by armandojaleo**
Add new and update types
Add new and update tests
Add new and update interfaces
Add new and update implementations

https://api-reference.checkout.com/#operation/requestCardMetadata

### CODE REVIEW

**Review by Armando-Ibarra — CHANGES_REQUESTED** [review]
Very good job here, a couple of comments.

Also, you need to add on `OAuthScope.cs` the new scope for metadata `[OAuthScope("vault:card-metadata")] VaultCardMetadata,`
And add into `SandboxTestFixture` during `case PlatformType.DefaultOAuth:` the added scope.

File: src/CheckoutSdk/MetaData/Card/IMetaDataCardClient.cs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Checkout.Metadata.Card
+{
+    public interface IMetadataCardClient
```
[the remark this answers is not in the document]
Comment (armandojaleo) [reply src/CheckoutSdk/MetaData/Card/IMetaDataCardClient.cs:6]: Changed

File: src/CheckoutSdk/MetaData/Card/MetaDataCardPayoutsTransactionsType.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+﻿using System.Runtime.Serialization;
+
+namespace Checkout.Metadata.Card
+{
+    public enum CardPayoutsTransferType
```
[the remark this answers is not in the document]
Comment (armandojaleo) [reply src/CheckoutSdk/MetaData/Card/MetaDataCardPayoutsTransactionsType.cs:5]: Changed

File: test/CheckoutSdkTest/MetaData/Card/MetaDataCardIntegrationTest.cs [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+            {
+                AddressLine1 = "CheckoutSdk.com",
+                AddressLine2 = "90 Tottenham Court Road",
+                City = "London",
+                State = "London",
+                Zip = "W1T 4TJ",
+                Country = CountryCode.GB
+            };
+
+            CardTokenRequest cardTokenRequest = new CardTokenRequest()
```
[the remark this answers is not in the document]
Comment (armandojaleo) [reply test/CheckoutSdkTest/MetaData/Card/MetaDataCardIntegrationTest.cs:248]: 👍

File: test/CheckoutSdkTest/MetaData/Card/MetaDataCardIntegrationTest.cs [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+        public MetadataCardIntegrationTest() : base(PlatformType.DefaultOAuth)
+        {
+        }
+        
+        [Fact]
+        private async Task ShouldRequestMetadataCardCardBasicExample()
+        {
+            MetadataCardRequest request = new MetadataCardRequest()
+            {
+                Number = CardNumberConstant
```
[the remark this answers is not in the document]
Comment (armandojaleo) [reply test/CheckoutSdkTest/MetaData/Card/MetaDataCardIntegrationTest.cs:26]: 👍

File: test/CheckoutSdkTest/MetaData/Card/MetaDataCardIntegrationTest.cs [hunk 5]
```diff
@@ -0,0 +1,265 @@
...
+                TokenId = $"tokError_{GetToken()}",
+                Format = MetadataCardFormatType.CardPayouts
+            };
+            
+            await CheckErrorItem(async () => await DefaultApi.MetadataCardClient().RequestCardMetadata(request),
+                PaymentSourceRequiredErrorConstant);
+        }
+        
+        [Fact]
+        private async Task ShouldRequestMetadataCardEmptyRequestError()
```
[the remark this answers is not in the document]
Comment (armandojaleo) [reply test/CheckoutSdkTest/MetaData/Card/MetaDataCardIntegrationTest.cs:226]: Perfect!

File: test/CheckoutSdkTest/MetaData/Card/MetaDataCardIntegrationTest.cs [hunk 6]
```diff
@@ -0,0 +1,265 @@
...
+            response.IssuerCountry.ShouldBeOfType<CountryCode>();
+            response.IssuerCountryName.ShouldNotBeNull();
+            response.ProductId.ShouldNotBeNull();
+            response.ProductType.ShouldNotBeNull();
+            response.HttpStatusCode.ShouldBe(200);
+            
+        }
+        
+        [Fact]
+        private async Task ShouldRequestMetadataCardNumberBasicExample()
```
[the remark this answers is not in the document]
Comment (armandojaleo) [reply test/CheckoutSdkTest/MetaData/Card/MetaDataCardIntegrationTest.cs:46]: 👍

File: test/CheckoutSdkTest/MetaData/Card/MetaDataCardIntegrationTest.cs [hunk 7]
```diff
@@ -0,0 +1,265 @@
...
+        {
+            MetadataCardRequest request = new MetadataCardRequest()
+            {
+                Bin = BinNumberConstant
```
[the remark this answers is not in the document]
Comment (armandojaleo) [reply test/CheckoutSdkTest/MetaData/Card/MetaDataCardIntegrationTest.cs:50]: 👍

File: test/CheckoutSdkTest/MetaData/Card/MetaDataCardIntegrationTest.cs [hunk 8]
```diff
@@ -0,0 +1,265 @@
...
+            response.Scheme.ShouldNotBeNull();
+            response.SchemeLocal.ShouldBeOfType<SchemeLocalType>();
+            response.CardType.ShouldBeOfType<CardType>();
+            response.CardCategory.ShouldBeOfType<CardCategory>();
+            response.Issuer.ShouldNotBeNull();
+            response.IssuerCountry.ShouldBeOfType<CountryCode>();
+            response.IssuerCountryName.ShouldNotBeNull();
+            response.ProductId.ShouldNotBeNull();
+            response.ProductType.ShouldNotBeNull();
+            response.HttpStatusCode.ShouldBe(200);
```
[the remark this answers is not in the document]
Comment (armandojaleo) [reply test/CheckoutSdkTest/MetaData/Card/MetaDataCardIntegrationTest.cs:65]: All tests methods refactored

File: test/CheckoutSdkTest/MetaData/Card/MetaDataCardIntegrationTest.cs [hunk 9]
```diff
@@ -0,0 +1,265 @@
...
+            
+        }
+        
+        [Fact]
+        private async Task ShouldRequestMetadataCardCardPayoutsExample()
```
[the remark this answers is not in the document]
Comment (armandojaleo) [reply test/CheckoutSdkTest/MetaData/Card/MetaDataCardIntegrationTest.cs:95]: 👍

File: test/CheckoutSdkTest/MetaData/Card/MetaDataCardIntegrationTest.cs [hunk 10]
```diff
@@ -0,0 +1,265 @@
...
+            response.HttpStatusCode.ShouldBe(200);
+            
+        }
+        
+        [Fact]
+        private async Task ShouldRequestMetadataCardTokenBasicExample()
+        {
+            MetadataCardRequest request = new MetadataCardRequest()
+            {
+                TokenId = $"tok_{GetToken()}",
```
[the remark this answers is not in the document]
Comment (armandojaleo) [reply test/CheckoutSdkTest/MetaData/Card/MetaDataCardIntegrationTest.cs:74]: Removed

File: src/CheckoutSdk/MetaData/Card/MetaDataCardPayoutsTransactionsType.cs [hunk 11]
```diff
@@ -0,0 +1,12 @@
+﻿using System.Runtime.Serialization;
+
+namespace Checkout.Metadata.Card
+{
+    public enum MetadataCardPayoutsTransactionsType
```
Comment (Armando-Ibarra) [inline src/CheckoutSdk/MetaData/Card/MetaDataCardPayoutsTransactionsType.cs:5]: How about calling this `CardPayoutsTransactionsType`?

Comment (armandojaleo) [reply src/CheckoutSdk/MetaData/Card/MetaDataCardPayoutsTransactionsType.cs]: 👍

**Review by Armando-Ibarra — APPROVED** [review]
Very good job

**Pull request merged**